src/arch/mod.rs

```diff
@@ -121,9 +127,11 @@ impl Type {
     pub fn ptr_type() -> IrType {
         IrType::int(CHAR_BIT * PTR_SIZE).expect("pointer size should be valid")
     }
-    pub fn struct_offset(&self, members: &[Symbol], member: InternedStr) -> u64 {
+    /// Get the offset of the given struct member.
+    pub fn struct_offset(&self, struct_type: &StructType, member: InternedStr) -> u64 {
```

Comment:
Do you know why this takes a struct type instead of just getting it from `self`? is it a remnant from earlier struct code?

Reply:
Good catch, this shouldn't take `self` at all; or rather, this should be in `impl StructType`, not `impl Type`.

Reply:
So Should I move this (and next_offset) to StructType?

Reply:
Yes, I think that would be better.